src/pages/index.js

```diff
@@ -29,60 +30,155 @@ const enableValidation = (config) => {
 
 enableValidation(validationConfig);
 
-//creating cards from the array
-const popupWithImage = new PopupWithImage(".modal_type_show-image");
+//api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "9e620f98-b2bc-4f4a-81bd-341fb1cf797f",
+  },
+});
 
-const handleImageClick = (card) => {
-  popupWithImage.open(card.link, card.name);
+//user
+api
+  .getUserInfo()
+  .then((userInfo) => {
+    //set user info
+    const user = new UserInfo(
+      ".explorer__name",
+      ".explorer__description",
+      ".explorer__photo"
+    );
+    user.setUserInfo({
+      name: userInfo.name,
+      about: userInfo.about,
+    });
+    user.setAvatar({ avatar: userInfo.avatar });
+    return user;
+  })
+  .then((user) => {
+    //cahnging user details
+    function saveProfileChanges(newUserData) {
+      profileEditPopup.savingDisplay(true);
+      user.setUserInfo(newUserData);
+      api.patchUserInfo(newUserData).then(() => {
+        profileEditPopup.close();
+        profileEditPopup.savingDisplay(false);
+      });
+    }
+
+    const profileEditPopup = new PopupWithForm(
+      ".modal_type_profile",
+      "profile-form",
+      saveProfileChanges,
+      "Save"
+    );
+    profileEditPopup.setEventListeners();
+    editBtn.addEventListener("click", () => {
+      formValidators["profile-form"].resetValidation();
+      const currentData = user.getUserInfo();
+      profileEditPopup.setInputValues(currentData);
+      profileEditPopup.open();
+    });
+
+    function saveAvatar(newAvatar) {
+      changeAvatarPopup.savingDisplay(true);
+      user.setAvatar(newAvatar);
+      api.patchAvatar(newAvatar).then(() => {
+        changeAvatarPopup.resetForm();
+        changeAvatarPopup.close();
+        changeAvatarPopup.savingDisplay(false);
+        formValidators["avatar-form"].disableButton();
+      });
+    }
+
+    const changeAvatarPopup = new PopupWithForm(
+      ".modal_type_avatar",
+      "avatar-form",
+      saveAvatar,
+      "Save"
+    );
+    changeAvatarPopup.setEventListeners();
+    avatarChangeBtn.addEventListener("click", () => {
+      changeAvatarPopup.open();
+    });
+  })
+  .catch((err) => console.error(err));
+
+const handleDeleteButton = (card) => {
+  popupConfirmDelete.open(card);
 };
-popupWithImage.setEventListeners();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const card = new Card(item, ".card", handleImageClick);
-      const cardElement = card.generateCard();
-      return cardElement;
-    },
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
+const handleDeleteConfirm = (card) => {
+  card.handleDeleteConfirm();
+  popupConfirmDelete.close();
+  api.deleteCard(card._id);
+};
+
+const handleLikeButton = (card) => {
+  card.isLiked === false ? api.setLike(card._id) : api.removeLike(card._id);
+  card.isLiked = !card.isLiked;
```

Comment:
* You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.